src/app/item-popup/ItemSocketsGeneral.tsx

```diff
@@ -19,25 +18,19 @@ interface Props {
   item: DimItem;
   /** minimal style used for loadout generator and compare */
   minimal?: boolean;
-  updateSocketComparePlug?(value: { item: DimItem; socket: DimSocket; plug: DimPlug }): void;
-  adjustedItemPlugs?: DimAdjustedItemPlug;
+  onPlugClicked?(value: { item: DimItem; socket: DimSocket; plug: DimPlug }): void;
 }
 
-export default function ItemSocketsGeneral({
-  item,
-  minimal,
-  updateSocketComparePlug,
-  adjustedItemPlugs,
-}: Props) {
+export default function ItemSocketsGeneral({ item, minimal, onPlugClicked }: Props) {
   const defs = useD2Definitions();
   const wishlistRoll = useSelector(wishListSelector(item));
   const [socketInMenu, setSocketInMenu] = useState<DimSocket | null>(null);
 
   const handleSocketClick = (item: DimItem, socket: DimSocket, plug: DimPlug, hasMenu: boolean) => {
     if (hasMenu) {
       setSocketInMenu(socket);
-    } else if (updateSocketComparePlug) {
-      updateSocketComparePlug({
+    } else if (onPlugClicked) {
```

Comment:
matter of taste but i like
```
    } else {
      onPlugClicked?.({
        item,
        socket,
        plug,
      });
```

because this `else if` isn't really related to the original `if (hasMenu`
it's more of an `else`. then an unrelated `if` to check if there's a function.